synapse/rest/client/v2_alpha/auth.py

```diff
@@ -144,7 +146,7 @@ def on_GET(self, request, stagetype):
                 ),
                 'sitekey': self.hs.config.recaptcha_public_key,
             }
-            html_bytes = html.encode("utf8")
+            html_bytes = ensure_binary(html)
```

Comment:
the fix here is to decide if RECAPTCHA_TEMPLATE is utf-encoded or not, and give it a `b` or `u` prefix accordingly - presumably the latter